grid.cpp

```diff
@@ -604,14 +599,15 @@ void balanceLoad(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid, S
 
          int prepareReceives {phiprof::initializeTimer("Preparing receives")};
          int receives = 0;
+         #pragma omp parallel for schedule(guided)
          for (unsigned int i=0; i<incoming_cells_list.size(); i++) {
             CellID cell_id=incoming_cells_list[i];
             SpatialCell* cell = mpiGrid[cell_id];
             if (cell_id % num_part_transfers == transfer_part) {
                receives++;
                // reserve space for velocity block data in arriving remote cells
                phiprof::Timer timer {prepareReceives};
```

Comment:
Tangent but horrible place for a timer, pull it out of the omp loop at least?...

Reply:
The timer is prepared outside the parallel region, just activated in there for each thread. AFAIK this is the intended way to timer in parallel regions.